### PR TITLE
Use Kawase Blur

### LINKED ISSUE #271: Blur Background setting causing significant GPU usage above certain settings
**Describe the bug**

When setting the `Blur background factor` setting high enough, the background blur effect will use a very significant amount of GPU resources (this is more apparent on integrated graphics).

**To Reproduce**

1. Open Task Manager and OBS Studio
2. Add a `Background Removal` filter to any source (if you haven't already).
3. Set `Blur Background Factor` to a high setting (settings as low as 6 max out my integrated graphics, while my discrete graphics can handle up to around 12 before lagging significantly)
4. Observe (in Task Manager) the High GPU usage

**Expected behavior**

The background blur does not use this much GPU time.

**Screenshots**

Before (tested using 39adf39) | After (latest release)
------------------------------|------------------------
before | after

Background blur factor was set to 50 (before) and 12 (after). These settings give about the same amount of blur when switching versions, from my testing.

**Log and Crash Report**

[2023-04-24 07-53-54.txt](https://github.com/royshil/obs-backgroundremoval/files/11305075/2023-04-24.07-53-54.txt)

[2023-04-24 07-55-51.txt](https://github.com/royshil/obs-backgroundremoval/files/11305074/2023-04-24.07-55-51.txt)

**Desktop (please complete the following information):**

 - OS: Windows 10 (22H2)
 - Browser: N/A
 - Plugin Version: Latest (v0.5.17)
 - OBS Version: Latest (v29.0.2)

**Additional context**

Commit 39adf39 and earlier do not seem to exhibit this behaviour, which suggests that the newly-introduced GPU blending in the latest release is actually performing worse than before

### PR DESCRIPTION
**Pull request by umireon**
Fixes #271

https://www.intel.com/content/www/us/en/developer/articles/technical/an-investigation-of-fast-real-time-gpu-based-image-blur-algorithms.html#:~:text=%C2%A0%22Kawase%20blur%22%20filter

### CODE REVIEW

File: src/background-filter.cpp [hunk 1]
```diff
@@ -554,6 +561,30 @@ static void filter_video_render(void *data, gs_effect_t *_effect)
 
   // Output the masked image
 
+  gs_texture_t *blurredTexture = gs_texture_create(width, height, GS_BGRA, 1, nullptr, 0);
+  gs_copy_texture(blurredTexture, gs_texrender_get_texture(tf->texrender));
+  for (int i = 0; i < (int)tf->blurBackground; i++) {
```
Comment (royshil) [inline src/background-filter.cpp]: we can move many things out of the loop...

File: src/background-filter.cpp [hunk 2]
```diff
@@ -554,6 +561,30 @@ static void filter_video_render(void *data, gs_effect_t *_effect)
 
   // Output the masked image
 
+  gs_texture_t *blurredTexture = gs_texture_create(width, height, GS_BGRA, 1, nullptr, 0);
```
Comment (royshil) [inline src/background-filter.cpp]: we should only perform any of this if `tf->backgroundBlur > 0`

**Review by royshil — CHANGES_REQUESTED** [review]
this looks good - but lets clean it up first

File: src/background-filter.cpp (hunk 1 above)
Comment (umireon) [reply src/background-filter.cpp]: Fixed

File: src/background-filter.cpp (hunk 2 above)
Comment (umireon) [reply src/background-filter.cpp]: Fixed!

**Review by royshil — APPROVED** [review]
nice! well done! go

**Comment by royshil** [comment]
@umireon  wait! we got a problem
works great in the filters screen
image
but in the main view....
image
we have a problem

**Comment by umireon** [comment]
On my side, everything works fine.

スクリーンショット 2023-04-25 23 31 21
スクリーンショット 2023-04-25 23 31 18

**Comment by royshil** [comment]
@umireon i'm still having a problem, and getting a strange error on the log
```
14:26:30.848: device_texture_create (D3D11): Failed to create 2D texture (80070057)
14:26:30.848: [obs-backgroundremoval] Failed to create alpha texture
14:26:30.853: device_texture_create (D3D11): Failed to create 2D texture (80070057)
14:26:30.853: [obs-backgroundremoval] Failed to create alpha texture
```
this error doesn't come up on the main version of the plugin.

so i think it's related to this change you're proposing here

**Comment by umireon** [comment]
@royshil Can you paste the full log here?

**Comment by umireon** [comment]
I could reproduced the problem on my side

スクリーンショット 2023-04-26 13 40 57

**Comment by umireon** [comment]
Fixed

**Review by royshil — APPROVED** [review]
confirmed to fix the problem - fire at will! 🚀

**Pull request merged**